fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java

```diff
@@ -361,18 +363,30 @@ public void sendVote(List<ElectionVote> votes) {
 
     try {
       PoPToken token = keyManager.getValidPoPToken(lao);
-      CastVote castVote = new CastVote(votes, election.getId(), lao.getId());
-      Channel electionChannel = election.getChannel();
+      CastVote<ElectionVote> castOpenBallotVote = null;
+      CastVote<ElectionEncryptedVote> castEncryptedVote = null;
+
+      // Construct the cast vote depending if the messages need to be encrypted or not
+      if (election.getElectionVersion() == ElectionVersion.OPEN_BALLOT) {
+        castOpenBallotVote = new CastVote<>(votes, election.getId(), lao.getId());
+      } else {
+        Log.d(TAG, "Encrypting the vote ...");
```

Comment:
Small detail but these logs are for debugging purposes. This looks more like communication to the user, in which case you should use a toast

Reply:
This seems more logical! dont know why I put a log there